### ISSUE TITLE
Xenon crash: `blockstack_lib::chainstate::coordinator::get_reward_cycle_info`

### ISSUE THREAD

**Issue by diwakergupta** [blockstack/stacks-blockchain#2240]
Stack trace:

```
   0: backtrace::backtrace::libunwind::trace
             at /Users/diwaker/.cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.50/src/backtrace/libunwind.rs:95
      backtrace::backtrace::trace_unsynchronized
             at /Users/diwaker/.cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.50/src/backtrace/mod.rs:66
      backtrace::backtrace::trace
             at /Users/diwaker/.cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.50/src/backtrace/mod.rs:53
   1: backtrace::capture::Backtrace::create
             at /Users/diwaker/.cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.50/src/capture.rs:164
      backtrace::capture::Backtrace::new
             at /Users/diwaker/.cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.50/src/capture.rs:128
   2: stacks_node::main::{{closure}}
             at /Users/diwaker/Vault/blockstack/stacks-blockchain/testnet/stacks-node/src/main.rs:57
   3: std::panicking::rust_panic_with_hook
             at /rustc/7eac88abb2e57e752f3302f02be5f3ce3d7adfb4/library/std/src/panicking.rs:581
   4: std::panicking::begin_panic_handler::{{closure}}
             at /rustc/7eac88abb2e57e752f3302f02be5f3ce3d7adfb4/library/std/src/panicking.rs:484
   5: std::sys_common::backtrace::__rust_end_short_backtrace
             at /rustc/7eac88abb2e57e752f3302f02be5f3ce3d7adfb4/library/std/src/sys_common/backtrace.rs:153
   6: rust_begin_unwind
             at /rustc/7eac88abb2e57e752f3302f02be5f3ce3d7adfb4/library/std/src/panicking.rs:483
   7: core::panicking::panic_fmt
             at /rustc/7eac88abb2e57e752f3302f02be5f3ce3d7adfb4/library/core/src/panicking.rs:85
   8: core::option::expect_failed
             at /rustc/7eac88abb2e57e752f3302f02be5f3ce3d7adfb4/library/core/src/option.rs:1226
   9: core::option::Option<T>::expect
             at /Users/diwaker/.rustup/toolchains/stable-x86_64-apple-darwin/lib/rustlib/src/rust/library/core/src/option.rs:346
      blockstack_lib::chainstate::burn::db::sortdb::<impl blockstack_lib::util::db::IndexDBConn<blockstack_lib::chainstate::burn::db::sortdb::SortitionHandleContext,blockstack_lib::chainstate::burn::db::sortdb::SortitionId>>::get_chosen_pox_anchor
             at /Users/diwaker/Vault/blockstack/stacks-blockchain/src/chainstate/burn/db/sortdb.rs:1802
  10: blockstack_lib::chainstate::coordinator::get_reward_cycle_info
             at /Users/diwaker/Vault/blockstack/stacks-blockchain/src/chainstate/coordinator/mod.rs:405
  11: blockstack_lib::chainstate::coordinator::ChainsCoordinator<T,N,U>::get_reward_cycle_info
             at /Users/diwaker/Vault/blockstack/stacks-blockchain/src/chainstate/coordinator/mod.rs:589
      blockstack_lib::chainstate::coordinator::ChainsCoordinator<T,N,U>::handle_new_burnchain_block
             at /Users/diwaker/Vault/blockstack/stacks-blockchain/src/chainstate/coordinator/mod.rs:534
  12: blockstack_lib::chainstate::coordinator::ChainsCoordinator<T,blockstack_lib::chainstate::coordinator::comm::ArcCounterCoordinatorNotices,blockstack_lib::chainstate::coordinator::OnChainRewardSetProvider>::run
             at /Users/diwaker/Vault/blockstack/stacks-blockchain/src/chainstate/coordinator/mod.rs:299
      stacks_node::run_loop::neon::RunLoop::start::{{closure}}
             at /Users/diwaker/Vault/blockstack/stacks-blockchain/testnet/stacks-node/src/run_loop/neon.rs:231
      std::sys_common::backtrace::__rust_begin_short_backtrace
             at /Users/diwaker/.rustup/toolchains/stable-x86_64-apple-darwin/lib/rustlib/src/rust/library/std/src/sys_common/backtrace.rs:137
  13: std::thread::Builder::spawn_unchecked::{{closure}}::{{closure}}
             at /Users/diwaker/.rustup/toolchains/stable-x86_64-apple-darwin/lib/rustlib/src/rust/library/std/src/thread/mod.rs:464
      <std::panic::AssertUnwindSafe<F> as core::ops::function::FnOnce<()>>::call_once
             at /Users/diwaker/.rustup/toolchains/stable-x86_64-apple-darwin/lib/rustlib/src/rust/library/std/src/panic.rs:308
      std::panicking::try::do_call
             at /Users/diwaker/.rustup/toolchains/stable-x86_64-apple-darwin/lib/rustlib/src/rust/library/std/src/panicking.rs:381
      std::panicking::try
             at /Users/diwaker/.rustup/toolchains/stable-x86_64-apple-darwin/lib/rustlib/src/rust/library/std/src/panicking.rs:345
      std::panic::catch_unwind
             at /Users/diwaker/.rustup/toolchains/stable-x86_64-apple-darwin/lib/rustlib/src/rust/library/std/src/panic.rs:382
      std::thread::Builder::spawn_unchecked::{{closure}}
             at /Users/diwaker/.rustup/toolchains/stable-x86_64-apple-darwin/lib/rustlib/src/rust/library/std/src/thread/mod.rs:463
      core::ops::function::FnOnce::call_once{{vtable.shim}}
             at /Users/diwaker/.rustup/toolchains/stable-x86_64-apple-darwin/lib/rustlib/src/rust/library/core/src/ops/function.rs:227
  14: <alloc::boxed::Box<F> as core::ops::function::FnOnce<A>>::call_once
             at /rustc/7eac88abb2e57e752f3302f02be5f3ce3d7adfb4/library/alloc/src/boxed.rs:1042
      <alloc::boxed::Box<F> as core::ops::function::FnOnce<A>>::call_once
             at /rustc/7eac88abb2e57e752f3302f02be5f3ce3d7adfb4/library/alloc/src/boxed.rs:1042
      std::sys::unix::thread::Thread::new::thread_start
             at /rustc/7eac88abb2e57e752f3302f02be5f3ce3d7adfb4/library/std/src/sys/unix/thread.rs:87
  15: __pthread_start

Dumping core for pid 60434
```

Full debug logs: https://www.dropbox.com/s/riivgnxbjeorlg1/xenon-miner-diwaker-2020-12-31.log.gz?dl=0

**Comment by lgalabru** [comment]
Addressed with #2241, tests still need be written.

**Comment by kantai** [comment]
Fixed in #2242
